### PR TITLE
Clarify the use of `:not` with invalid selectors

### LINKED ISSUE #24277: inaccurate example on the :not() page
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/:not

### What specific section or headline is this issue about?

Description

### What information was incorrect, unhelpful, or incomplete?

last bullet:
> If any selector passed to the `:not()` pseudo-class is invalid or not supported by the browser, the whole rule will be invalidated. The effective way to overcome this behavior is to use [`:is`](https://developer.mozilla.org/en-US/docs/Web/CSS/:is) pseudo-class, which accepts a forgiving selector list. For example `:not(.foo, :invalid-pseudo-class)` will invalidate a whole rule, but `:is(:not(.foo), :not(:invalid-pseudo-class))` will match any element that isn't `.foo`.

### What did you expect to see?

the example is inaccurate, as `:not(.foo, :invalid-pseudo-class)` matches NEITHER `.foo` NOR `:invalid-pseudo-class`
whereas `:is(:not(.foo), :not(:invalid-pseudo-class))` matches anything that simply isn't `.foo` (disregarding the second condition)
it will also match elements that weren't intended to initially, thus isn't a very good example of overcoming the issue.. :)

### Do you have any supporting links, references, or citations?

made a [pen example](https://codepen.io/michal-amichai/pen/oNMRyWG)

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/_colon_not`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/:not
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/_colon_not/index.md
* Last commit: https://github.com/mdn/content/commit/26a33b20edc0facf5a7b17ad0bf3190d847e366e
* Document last modified: 2022-12-15T20:42:42.000Z

</details>

### PR DESCRIPTION
**Pull request by tlylt**
### Description
- Improve the explanation to mention that `html` and `body` may be selected unintentionally (it's mentioned in pt 4 but would be good to repeat here)
- Add example showing the effect of using `:not` with invalid selector and how to use `:is` + `:not`

### Motivation
Clarify and add examples.

### Additional details
nil

### Related issues and pull requests
Fixes #24277

### CODE REVIEW

File: files/en-us/web/css/_colon_not/index.md [hunk 1]
```diff
@@ -97,6 +97,41 @@ h2 :not(span.foo) {
...
+<p class="foo">Hello World with .foo</p>
+<p class="bar">Hello World with .bar</p>
+```
+
+#### CSS
+
+```css
+/* Invalid rule, does nothing */
+:not(.foo, :invalid-pseudo-class) {
+  color: red;
```
Comment (estelle) [inline files/en-us/web/css/_colon_not/index.md:117]:
```suggestion
  color: red;
  font-style: italic;
```

color blindness: don't want to rely on color alone.

File: files/en-us/web/css/_colon_not/index.md [hunk 2]
```diff
@@ -97,6 +97,41 @@ h2 :not(span.foo) {
...
+}
+
+/* Select all elements without the `foo` class, may include <html> and <body> elements */
+:is(:not(.foo), :not(:invalid-pseudo-class)) {
+  text-decoration: underline;
+}
+
+/* Select all <p> elements without the `foo` class */
+p:is(:not(.foo), :not(:invalid-pseudo-class)) {
+  color: green;
```
Comment (estelle) [inline files/en-us/web/css/_colon_not/index.md:127]:
```suggestion
  color: green;
  border-top: dotted thin currentcolor;
```

**Review by estelle — COMMENTED** [review]
Thank you.

I made edits to the examples: we don't want to use colors alone to demonstrate features for accessibility reasons. Also, if the red did work, it would be superceded by green, and everything in the example is a p. Perhaps remove the p from the green example and add it to the red to change which one has greater specificity?

**Comment by tlylt** [comment]
Thanks @estelle for the feedback, I have incorporated them.

> I made edits to the examples: we don't want to use colors alone to demonstrate features for accessibility reasons.

Sure, will take note of that in the future.

> Also, if the red did work, it would be superceded by green, and everything in the example is a p. Perhaps remove the p from the green example and add it to the red to change which one has greater specificity?

the p for the green example shows how it does not select the html and body element by accident. I have changed the first paragraph to a div instead so that it won't be affected by the last rule. Hope that's ok.

**Review by estelle — APPROVED** [review]
Thank you!

**Pull request merged**